docs/conf.py

```diff
@@ -47,6 +49,9 @@ def _ignore_offscreen_run():
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ["_templates"]
 
+# Just let autosummary produce a new version each time
+shutil.rmtree(os.path.join(os.path.dirname(__file__), "_autosummary"), True)
+
```

Comment:
@almarklein Could you elaborate why we want to delete the cache each time we build?

Autosummary already has a mechanism of invalidating the generated files (it compares last modified timestamps of the source and the generated file), so from my perspective, it looks like this mostly adds build time.

Reply:
> Autosummary already has a mechanism of invalidating the generated files

If so, it does not work properly, because I was seeing all kinds of warnings from old modules that have changed at some point. Preferably I would delete that dir on `make clean`, not sure how hard it'd be to make that work.

Reply:
Yes, if you modify the names of classes or modules you will generate new pages (since each class and each module gets its own doc page). autosummary doesn't delete files (not sure why), so the files matching the old names will linger and create warnings.

However, if you do normal coding and modify/update the docstrings or narrative docs then this caching saves you a couple of seconds each time you rebuild to inspect changes.